frc2609/Crescendo-2024
# Merge Changes from Waterloo

* Improve Limelight:
  * Add side Limelight
  * Improve Limelight Standard Deviations calculation (still requires adjustment (and distance equation needs replacement))
* Speed up autos
* Add SmartDashboard layout
* Add USB webcam for intake
* Added Drive Team Prematch Checklist
* Added several new autos
  * "5912 waterloo" 2 piece (tested, questionably working)
  * "Alt Four Piece" 3 piece (untested)
  * "Amp Side 3 Piece"
  * "Source Side Disruption" knock pieces around (untested)
    * Also a continuous path for "Source Side Disruption"
* Modified existing autos:
  * Slowed down four piece (still nonfunctional)
* Adjusted a variety of absolute encoder offsets
* Increased joystick deadband (new xbox controllers)
* Added LEDs!
* Use getVelocity() from Drive
* Modified controls:
  * Added presets to throw note across field
  * Added buttons to operator to update odometry according to specific limelight
* Rotate towards AprilTag when odometry is invalid or hasn't been updated for 0.5s (as suggested by Cybercavs)
* Adjust shooter height offset
* Add function to check if drive odometry is outside of the field
* Log target heading of swerve controller
* Set LEDs according to intake
* Tune shooter angle
* Improve getAngle functions in ShooterAngle
* Use absolute shooter angle in Visualizer
